### ISSUE TITLE
Gitlab service installation fails due to an directory permission error while extracting a tar file

### ISSUE THREAD

**Issue by toogley** [NixOS/nixpkgs#28945]
## Issue description

Gitlab service installation fails due to an directory permission error while extracting a tar file.

### Steps to reproduce

* put that into configuration.nix: 
```
services = {
    gitlab = {
      databasePassword = "test";
      secrets = {
        secret = "asrjkwbrewkbfskjdfdsfkjrewbgtregtrenmlgkangösdfsdfsädfsd,fdsfoierg9834g3l";
	db = "ewrwerewrwekjbrasbdfkdsfbsdgmfdlhgfhtrihtrotigfhdöhgfhjöfgabfsabfwafawbfesgs";
	otp = "resfrsefj8432t58439t983ngdsfngdfogndgnfdiogadpomcvnbövcpnpbvnxün<üfnüernhpjnh";
      };
      enable = true;
      host = "nixos";
      port = 80;

    };
}
```
* run `sudo nixos-rebuild switch`
* result:

```
building Nix...
building the system configuration...
these derivations will be built:
  /nix/store/qsb2mrhfdyjw7830mfdzi8nbrbc66d8x-gitlab-8.17.6.drv
  /nix/store/pamzcbxfk9ndvcy2f18alppz9hgcmb2w-gitlab-rake.drv
  /nix/store/h97c0csy14yakzpy138hfn9w8mpz4pbn-system-path.drv
  /nix/store/2a4q9bp2v5i83b3rqz771cmmm2byi7qs-unit-polkit.service.drv
  /nix/store/5sw9plkym0fxbsj8gpbhmpdrsbdf43cp-unit-systemd-fsck-.service.drv
  /nix/store/nlxaxs4vl4r71m79c1l949ahcvzq4w1v-dbus-1.drv
  /nix/store/j9f17jz73zsfs234j5h08ps4fwz7qldb-unit-dbus.service.drv
  /nix/store/nvfd8mhpa2y7db4ps4pi02m9ng3jjjbk-unit-gitlab-workhorse.service.drv
  /nix/store/fcnzg7ss9a1i7cslf0sdxl7v627b8xf7-unit-script.drv
  /nix/store/qy2rndgxrnhg6j7pb22jzwvhlasnj5ma-unit-gitlab.service.drv
  /nix/store/vzw43g2zf6q1cz5mdhph36zyalz5ksch-unit-gitlab-sidekiq.service.drv
  /nix/store/fgxvki4qkzavnfa3vi22n29wrk29vali-system-units.drv
  /nix/store/li02m8n9abdxgvls79zaf61xk54ll4ix-user-units.drv
  /nix/store/2r0pr0h5dqb1by306a5s7p7zz5ic2rnl-etc.drv
  /nix/store/y25ix2vgj0mghi67k9mpr6qgv6kjywsv-nixos-system-nixos-17.03.1775.56da88a298.drv
building path(s) ‘/nix/store/0q16jqsnsgm0pm77p2n4jhbhcnkhs07g-gitlab-8.17.6’
unpacking sources
unpacking source archive /nix/store/lfapnnmck4jg01p89bhzqphvvy8bc1b8-gitlabhq-v8.17.6-src
source root is gitlabhq-v8.17.6-src
patching sources
applying patch /nix/store/7jrncyp165022g49nybb4b03x3gi3f3p-remove-hardcoded-locations.patch
patching file config/environments/production.rb
patching file config/gitlab.yml.example
Hunk #1 succeeded at 512 (offset 36 lines).
patching file config/initializers/1_settings.rb
Hunk #1 succeeded at 201 (offset 9 lines).
Hunk #2 succeeded at 400 (offset 38 lines).
patching file lib/gitlab/logger.rb
patching file lib/gitlab/uploads_transfer.rb
Hunk #1 succeeded at 1 with fuzz 2 (offset -28 lines).
patching file lib/tasks/gitlab/check.rake
applying patch /nix/store/z8ndqjxn7lc8mcmh6i1dxbxn6xav610x-nulladapter.patch
patching file Gemfile
Hunk #1 succeeded at 117 with fuzz 1 (offset -1 lines).
Hunk #2 succeeded at 351 with fuzz 2.
patching file Gemfile.lock
Hunk #2 succeeded at 416 with fuzz 2 (offset 18 lines).
Hunk #3 succeeded at 431 with fuzz 2 (offset 22 lines).
Hunk #4 succeeded at 826 (offset 24 lines).
configuring
no configure script, doing nothing
building
tar: ./opt/gitlab/embedded/bin/ksu: Cannot change mode to rwsr-xr-x: Operation not permitted
tar: Exiting with failure status due to previous errors
dpkg-deb: error: subprocess tar returned error exit status 2
builder for ‘/nix/store/qsb2mrhfdyjw7830mfdzi8nbrbc66d8x-gitlab-8.17.6.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/pamzcbxfk9ndvcy2f18alppz9hgcmb2w-gitlab-rake.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/vzw43g2zf6q1cz5mdhph36zyalz5ksch-unit-gitlab-sidekiq.service.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/nvfd8mhpa2y7db4ps4pi02m9ng3jjjbk-unit-gitlab-workhorse.service.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/qy2rndgxrnhg6j7pb22jzwvhlasnj5ma-unit-gitlab.service.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/h97c0csy14yakzpy138hfn9w8mpz4pbn-system-path.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/fgxvki4qkzavnfa3vi22n29wrk29vali-system-units.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/y25ix2vgj0mghi67k9mpr6qgv6kjywsv-nixos-system-nixos-17.03.1775.56da88a298.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/y25ix2vgj0mghi67k9mpr6qgv6kjywsv-nixos-system-nixos-17.03.1775.56da88a298.drv’ failed
```

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...) 17.03.1775.56da88a298 (Gorilla)
* Nix version: (run `nix-env --version`) nix-env (Nix) 1.11.13
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`) 17.03.1775.56da88a298
* Sandboxing enabled: (run `grep build-use-sandbox /etc/nix/nix.conf`) build-use-sandbox = false

**Comment by globin** [comment]
This is already fixed on the 17.03 branch and also doesn't occur anymore on master/17.09 due to the update to gitlab 9

**Comment by toogley** [comment]
@globin how do i update that? i have not installed gitlab through nix-env -i or environment.systemPackages in configuration.nix. Therefore, i don't think nix-env -u works here. And i can't seem to find other solutions. nix-channel --update also doesn't seem to work.
